tests/test_toa_reader.py

```diff
@@ -132,3 +132,17 @@ def test_tttai():
     m = get_model(parstr)
     y = toa.get_TOAs("test1.tim", model=m)
     assert y.clock_corr_info["include_bipm"] == False
+
+
+def test_toa_merge():
+    filenames = ["NGC6440E.tim", "testtimes.tim", "parkes.toa"]
+    toas = [toa.get_TOAs(ff) for ff in filenames]
+    nt = toa.merge_TOAs(toas)
+    assert len(nt.observatories) == 3
+    assert nt.table.meta["filename"] == filenames
+    assert nt.ntoas == 78
+    # now test a failure if ephems are different
```

Comment:
This could be made less dependent on details of the tim files by checking the length against the sum of the lengths.